### PR TITLE
Removing unused pkgs and files from backend and frontend | Adding code format check for backend and frontend

### PR DESCRIPTION
**Pull request by imrajdas**
Changes:
* Removing unused frontend and backend pkgs
* Adding checks to unused pkg  and code format of backend and frontend

```
Unused dependencies
* @auth0/auth0-react
* @pmmmwh/react-refresh-webpack-plugin
* @svgr/webpack
* @testing-library/jest-dom
* @testing-library/react
* @testing-library/user-event
* @types/jest
* @types/node
* @typescript-eslint/eslint-plugin
* @typescript-eslint/parser
* babel-eslint
* babel-jest
* babel-plugin-named-asset-import
* babel-preset-react-app
* bfj
* camelcase
* case-sensitive-paths-webpack-plugin
* dotenv
* dotenv-expand
* eslint-config-react-app
* eslint-plugin-flowtype
* eslint-plugin-import
* eslint-plugin-jest
* eslint-plugin-jsx-a11y
* eslint-plugin-react
* eslint-plugin-react-hooks
* eslint-plugin-testing-library
* eslint-webpack-plugin
* fs-extra
* identity-obj-proxy
* jest
* jest-circus
* jest-resolve
* jest-watch-typeahead
* pnp-webpack-plugin
* postcss-loader
* postcss-normalize
* postcss-preset-env
* postcss-safe-parser
* prop-types
* react-app-polyfill
* react-chips
* react-dev-utils
* react-graph-vis
* react-modal
* react-refresh
* react-vis
* resolve
* resolve-url-loader
* semver
* terser-webpack-plugin
* ts-pnp
* typescript
* url-loader
* web-vitals
* webpack-dev-server
* webpack-manifest-plugin
* workbox-webpack-plugin
Unused devDependencies
* @babel/plugin-syntax-jsx
* babel-plugin-styled-components
* husky
* lint-staged
* react-hot-loader
* webpack-cli
```

Test Pipeline- https://github.com/SigNoz/signoz/actions/runs/1064358763

Screenshot 2021-07-25 at 3 33 05 PM

Signed-off-by: rajdas98 <mail.rajdas@gmail.com>

### CODE REVIEW

**Comment by rajdas98** [comment]
cc: @ankitnayan

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,43 @@ FRONTEND_DOCKER_IMAGE ?= frontend
 FLATTERNER_DOCKER_IMAGE ?= query-service
 QUERY_SERVICE_DOCKER_IMAGE ?= flattener-processor
 
-all: build-push-frontend build-push-query-service build-push-flattener
+all: frontend-service-checks backend-services-checks build-push-frontend build-push-query-service build-push-flattener
+
+frontend-service-checks:
+	@echo "------------------"
+	@echo "--> Checking unused imports of signoz's frontend [depcheck]"
```
Comment (nidhi-tandon) [inline Makefile:20]: What is the motive for frontend checks? What do we want to achieve? 🤔

Comment (rajdas98) [reply Makefile:20]: In this PR, I have removed all the unused imports from the front end. So, this check will restrict developers from pushing unused dependencies to the master branch.

This is a  depcheck that will list all unused dependencies.

**Comment by palashgdev** [comment]
@rajdas98 there are some dependencies that are removed which might break the frontend and some libraries will be used in future 

1. @testing-library/react
2. @testing-library/jest-dom
3. @testing-library/react
4. @testing-library/user-event
5. dotenv
6. jest
7. typescript
8. husky
9. webpack-dev-server
10. webpack-cli
10. web-vitals
11. husky
12. react-hot-loader

these are the rough list of the packages.

**Comment by rajdas98** [comment]
> @rajdas98 there are some dependencies that are removed which might break the frontend and some libraries will be used in future
> 
> 1. @testing-library/react
> 2. @testing-library/jest-dom
> 3. @testing-library/react
> 4. @testing-library/user-event
> 5. dotenv
> 6. jest
> 7. typescript
> 8. husky
> 9. webpack-dev-server
> 10. webpack-cli
> 11. web-vitals
> 12. husky
> 13. react-hot-loader
> 
> these are the rough list of the packages.

Thanks, @palashgdev for the information. So, Are we using these deps in our codebase? I have used the `depcheck` tool to identify the unused deps. If it is needed in future, then we can add it later also.

**Pull request closed without merging**